spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Graph.scala

```diff
@@ -297,8 +297,7 @@ class Graph[T: ClassTag](val inputs : Seq[ModuleNode[T]],
   /**
    * Execution plan
    */
-  private val forwardNodes = backGraph.DFS
-    .filterNot(_.element.isInstanceOf[ControlDependency[T]]).toArray
+  private val forwardNodes = backGraph.DFS.toArray
```

Comment:
do we need to update forwardSchduler, check roots and getForwardExecutions? saveGraphTopology?

Reply:
why remove the control dependency filter here?

Reply:
This is a subtle point. If a ControlDependency node's in-degree is zero, then it's children can never be scheduled.

Reply:
why a controlDependency node in-degree is zero?